### PR TITLE
[Store onboarding] Network and Yosemite to fetch tasks

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Part of: #8892 

## Description
Adds the Network and Yosemite code to fetch store onboarding tasks from the server. 

Apologies for the 500+ changes. It is from the lengthy mock JSON files. 

## Testing instructions
CI passing should be enough. 

## Screenshots
NA

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Networking/Networking/Mapper/StoreOnboardingTaskListMapper.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    func map(response: Data) throws -> [StoreOnboardingTask] {
+        let decoder = JSONDecoder()
+        let taskGroup: [StoreOnboardingTaskGroup]
+
+        do {
+            taskGroup = try decoder
+                .decode(StoreOnboardingTaskEnvelope.self, from: response)
+                .group
+        } catch {
+            taskGroup = try decoder.decode([StoreOnboardingTaskGroup].self, from: response)
```
Comment (jaclync) [inline Networking/Networking/Mapper/StoreOnboardingTaskListMapper.swift:15]: is this to support REST API? 🎉

File: Networking/Networking/Model/StoreOnboardingTask.swift [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    }
+}
+
+public extension StoreOnboardingTask {
+    enum TaskType: Decodable, Equatable {
+        case addFirstProduct
+        case launchStore
+        case customizeDomains
+        case payments
+        case unsupported
```
Comment (jaclync) [inline Networking/Networking/Model/StoreOnboardingTask.swift]: non-blocking: do you think we can keep the value of the task for easier debugging/development? similar to how [`ProductType` can take in a custom type](https://github.com/woocommerce/woocommerce-ios/blob/faebeb54d4c06f40b730608ab3ebe65e2023c50b/Networking/Networking/Model/Product/ProductType.swift#L18)

File: Networking/NetworkingTests/Remote/StoreOnboardingTasksRemoteTests.swift [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    func test_it_returns_only_the_store_setup_tasks_from_response() async throws {
+        // Given
+        let remote = StoreOnboardingTasksRemote(network: network)
+        network.simulateResponse(requestUrlSuffix: "onboarding/tasks", filename: "store-onboarding-tasks")
+
+        // When
+        let tasks = try await remote.loadOnboardingTasks(siteID: 0)
+
+        // Then
+        XCTAssertEqual(tasks.count, 8)
```
Comment (jaclync) [inline Networking/NetworkingTests/Remote/StoreOnboardingTasksRemoteTests.swift:29]: nit: might be nice asserting on the whole task list including the type and `isComplete` state.

**Review by jaclync — APPROVED** [review]
LGTM 🚀

File: Networking/Networking/Mapper/StoreOnboardingTaskListMapper.swift (hunk 1 above)
Comment (selanthiraiyan) [reply Networking/Networking/Mapper/StoreOnboardingTaskListMapper.swift:15]: Yes. This is needed as we decided to show the onboarding tasks for self-hosted sites as well.

File: Networking/Networking/Model/StoreOnboardingTask.swift (hunk 2 above)
Comment (selanthiraiyan) [reply Networking/Networking/Model/StoreOnboardingTask.swift]: Good idea. Stored the `id` as string in https://github.com/woocommerce/woocommerce-ios/pull/9017/commits/9e0152583a84f8f191ca4db66434f548f99443f3

File: Networking/NetworkingTests/Remote/StoreOnboardingTasksRemoteTests.swift (hunk 3 above)
Comment (selanthiraiyan) [reply Networking/NetworkingTests/Remote/StoreOnboardingTasksRemoteTests.swift:29]: Done in https://github.com/woocommerce/woocommerce-ios/pull/9017/commits/3f7b08cee3e951b5c2d4a20cc65e96c60d7a75be

**Pull request merged**